packages/@azure/servicebus/data-plane/test/topicFilters.spec.ts

```diff
@@ -55,15 +55,15 @@ async function beforeEachTest(): Promise<void> {
   }
 
   ns = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
-  topicClient = getSenderClient(ns, ClientType.TopicFilterTestTopic) as TopicClient;
-  subscriptionClient = getReceiverClient(
+  topicClient = (await getSenderClient(ns, ClientType.TopicFilterTestTopic)) as TopicClient;
+  subscriptionClient = (await getReceiverClient(
     ns,
     ClientType.TopicFilterTestSubscription
-  ) as SubscriptionClient;
-  defaultSubscriptionClient = getReceiverClient(
+  )) as SubscriptionClient;
+  defaultSubscriptionClient = (await getReceiverClient(
     ns,
     ClientType.TopicFilterTestDefaultSubscription
-  ) as SubscriptionClient;
+  )) as SubscriptionClient;
 
   await purge(defaultSubscriptionClient);
```

Comment:
As we create new resources here (and they will be empty) do we still need this call?

Reply:
We were discussing an option to not go through the whole setup/teardown when running the tests locally, in which case this purge will be necessary.

This option can be helpful as it removes the steps to create an AAD, then an application, then register the application, then give it access to the namespace.

We can have this option be driven though an environment variable. If set, then we skip the whole delete/create step.

Reply:
Good point.
Using the same option, we can do `purge` only if we test without creating/deleting entities.